File: credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int checksum(long c_no);                            // function to validate Luhn's Algorithm
int no_len(long c_no);                              // function to return number of digits of a number

int main(void)
{
    long c_no = get_long("Number : ");
    if (checksum(c_no) == 1)
    {
        if (no_len(c_no) == 15)                     // check 15 digit numbers for possible validity
        {
            if (c_no / 10000000000000 == 34 || c_no / 10000000000000 == 37)
            {
                printf("AMEX\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else if (no_len(c_no) == 16)                // check 16 digit numbers for possible validity
        {
            if (c_no / 100000000000000 >= 51 && c_no / 100000000000000 <= 55)
            {
                printf("MASTERCARD\n");
            }
            else if (c_no / 1000000000000000 == 4)
            {
                printf("VISA\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else if (no_len(c_no) == 13)                // check 13 digit numbers for possible validity
        {
            if (c_no / 1000000000000 == 4)
            {
                printf("VISA\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

int checksum(long c_no)                                  // function to validate Luhn's Algorithm
{
    int s_eve = 0, s_odd = 0, temp = 0;
    while (c_no != 0)
    {
        s_odd += c_no % 10;
        c_no /= 10;
        temp = (c_no % 10) * 2;
        while (temp != 0)
        {
            s_eve += temp % 10;
            temp /= 10;
        }
        c_no /= 10;
    }

    if ((s_odd + s_eve) % 10 == 0)
    {
        return 1;
    }
    else
    {
        return 0;
    }
}

int no_len(long c_no)                                     // function to return number of digits of a number
{
    int len = 0;
    while (c_no != 0)
    {
        len++;
        c_no /= 10;
    }
    return len;
}

File: credit.py
from cs50 import get_int
from sys import exit

# Main Function


def main():

    # Get the credit card number from the user
    number = get_int("Number: ")
    number = str(number)

    # Check for Validity of the Credit Card Number
    validity = checksum(number)
    if validity == False:
        print("INVALID")
        exit(0)

    # Check for AMEX
    if len(number) == 15:

        if number[0] == '3':

            if number[1] == '4' or number[1] == '7':
                print("AMEX")

            else:
                print("INVALID")

        else:
            print("INVALID")

    # Check for MASTERCARD or VISA
    elif len(number) == 16:

        if number[0] == '5':

            if number[1] in ('1', '2', '3', '4', '5'):
                print("MASTERCARD")

            else:
                print("INVALID")

        elif number[0] == '4':
            print("VISA")

        else:
            print("INVALID")

    # Check for VISA
    elif len(number) == 13:

        if number[0] == '4':
            print("VISA")

        else:
            print("INVALID")

    else:
        print("INVALID")


# Function to check validity of a credit number
def checksum(number):
    # Declaration of variables
    num = int(number)
    evesum = 0
    oddsum = 0

    # Getting one number at a time
    while num != 0:
        evesum += num % 10
        num //= 10

        temp = (num % 10) * 2
        while temp != 0:
            oddsum += temp % 10
            temp //= 10
        num //= 10

    # Check for validity
    if (oddsum + evesum) % 2 == 0:
        return True
    else:
        return False


main()